react/features/base/react/components/web/MultiSelectAutocomplete.js

```diff
@@ -0,0 +1,319 @@
+                    isDisabled = { isDisabled }
+                    isOpen = { this.state.isOpen }
+                    items = { this.state.items }
+                    noMatchesFound = { noMatchesFound }
+                    onFilterChange = { this._onFilterChange }
+                    onRemoved = { this._onSelectionChange }
+                    onSelected = { this._onSelectionChange }
+                    placeholder = { placeholder }
+                    selectedItems = { this.state.selectedItems }
+                    shouldFitContainer = { shouldFitContainer }
+                    shouldFocus = { shouldFocus } />
+                { this._renderLoadingIndicator() }
+                { this._renderError() }
+            </div>
+        );
+    }
+
+    /**
+     * Sets the state and sends a query on filter change.
+     *
+     * @param {string} filterValue - The filter text value.
+     * @private
+     * @returns {void}
+     */
+    _onFilterChange(filterValue) {
+        this.setState({
+            // Clean the error if the filterValue is empty.
+            error: this.state.error && Boolean(filterValue),
+            filterValue,
+            isOpen: Boolean(this.state.items.length) && Boolean(filterValue),
+            items: filterValue ? this.state.items : []
+        });
+        if (filterValue) {
+            this._sendQuery(filterValue);
+        }
+    }
+
+    /**
+     * Retries the query on retry.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onRetry() {
+        this._sendQuery(this.state.filterValue);
+    }
+
+    /**
+     * Updates the selected items when a selection event occurs.
+     *
+     * @param {Object} item - The selected item.
+     * @private
+     * @returns {void}
+     */
+    _onSelectionChange(item) {
+        const existing
+            = this.state.selectedItems.find(k => k.value === item.value);
+        let selectedItems = this.state.selectedItems;
+
+        if (existing) {
+            selectedItems = selectedItems.filter(k => k !== existing);
+        } else {
+            selectedItems.push(item);
+        }
+        this.setState({
+            isOpen: false,
+            selectedItems
+        });
+
+        if (this.props.onSelectionChange) {
+            this.props.onSelectionChange(selectedItems);
+        }
+    }
+
+    /**
+     * Renders the error UI.
+     *
+     * @returns {*}
+     */
+    _renderError() {
+        if (!this.state.error) {
+            return null;
+        }
+        const content = ( // eslint-disable-line no-extra-parens
+            <div className = 'autocomplete-error'>
+                <InlineDialogFailure
+                    onRetry = { this._onRetry } />
+            </div>
+        );
+
+        return (
+            <AKInlineDialog
+                content = { content }
+                isOpen = { true } />
+        );
+    }
+
+    /**
+     * Renders the loading indicator.
+     *
+     * @returns {*}
```

Comment:
I think the preferred documentation here has been something like {ReactElement|null}.